src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -717,6 +717,27 @@ public void stringNullAndNullWorksToo() {
         assertEquals("null", null);
     }
 
+    private static class NullToString {
+        @Override
+        public String toString() {
+            return null;
+        }
+    }
+
+    @Test
+    public void nullToString() {
+        try {
+            assertEquals(new NullToString(), new NullToString());
+        } catch (AssertionError e) {
+            assertEquals("expected: org.junit.tests.assertion.AssertionTest$NullToString<null> but "
+                            + "was: org.junit.tests.assertion.AssertionTest$NullToString<null>",
+                    e.getMessage());
+            return;
+        }
+
+        fail("Failed on assertion.");
```

Comment:
Well, the same holds true for all the other tests in this class, right? If so, please remove the `return` and the `fail()` statements.